build-system/tasks/dist.js

```diff
@@ -231,174 +224,135 @@ async function prebuild() {
-    gulp
-      .src(htmlPath)
-      .pipe(file('experiments.cdn.html', minHtml))
-      .pipe(gulp.dest('dist.tools/experiments/'))
-  );
+  await fs.ensureDir(htmlDestDir);
+  await fs.writeFile(`${htmlDestDir}/experiments.cdn.html`, minHtml);
+  await fs.ensureDir(expDir);
+  await fs.copy(htmlSrcPath, `${htmlDestDir}/${path.basename(htmlSrcPath)}`);
 
   // Build JS.
-  const js = fs.readFileSync(jsPath, 'utf8');
+  const jsDir = 'build/experiments/';
+  await fs.ensureDir(jsDir);
+  const js = await fs.readFile(jsSrcPath, 'utf8');
   const builtName = 'experiments.max.js';
-  return toPromise(
-    gulp
-      .src(path + '/*.js')
-      .pipe(file(builtName, js))
-      .pipe(gulp.dest('build/experiments/'))
-  );
+  await fs.writeFile(`${jsDir}/${builtName}`, js);
+  const jsFiles = globby.sync(`${expDir}/*.js`);
+  for (const jsFile of jsFiles) {
+    await fs.copy(jsFile, `${jsDir}/${path.basename(jsFile)}`);
+  }
 }
 
 /**
  * Build "Login Done" page.
- * @return {!Promise}
  */
-function preBuildLoginDone() {
-  return preBuildLoginDoneVersion('0.1');
+async function preBuildLoginDone() {
+  await preBuildLoginDoneVersion('0.1');
 }
 
 /**
  * Build "Login Done" page for the specified version.
- *
  * @param {string} version
- * @return {!Promise}
  */
-function preBuildLoginDoneVersion(version) {
-  const path = `extensions/amp-access/${version}/`;
-  const buildDir = `build/all/amp-access-${version}/`;
-  const htmlPath = path + 'amp-login-done.html';
-  const jsPath = path + 'amp-login-done.js';
+async function preBuildLoginDoneVersion(version) {
+  const srcDir = `extensions/amp-access/${version}`;
+  const buildDir = `build/all/amp-access-${version}`;
+  const htmlPath = `${srcDir}/amp-login-done.html`;
+  const jsPath = `${srcDir}/amp-login-done.js`;
 
   // Build HTML.
-  const html = fs.readFileSync(htmlPath, 'utf8');
+  const html = await fs.readFile(htmlPath, 'utf8');
   const minJs = `https://${hostname}/v0/amp-login-done-${version}.js`;
   const minHtml = html
     .replace(`../../../dist/v0/amp-login-done-${version}.max.js`, minJs)
     .replace(`../../../dist/v0/amp-login-done-${version}.js`, minJs);
   if (minHtml.indexOf(minJs) == -1) {
     throw new Error('Failed to correctly set JS in login-done.html');
   }
-
-  mkdirSync('dist');
-  mkdirSync('dist/v0');
-
-  fs.writeFileSync('dist/v0/amp-login-done-' + version + '.html', minHtml);
+  await fs.ensureDir('dist/v0');
+  await fs.writeFile(`dist/v0/amp-login-done-${version}.html`, minHtml);
 
   // Build JS.
-  const js = fs.readFileSync(jsPath, 'utf8');
-  const builtName = 'amp-login-done-' + version + '.max.js';
-  return toPromise(
-    gulp
-      .src(path + '/*.js', {base: path})
-      .pipe(file(builtName, js))
-      .pipe(gulp.dest(buildDir))
-  );
+  const js = await fs.readFile(jsPath, 'utf8');
+  const builtName = `amp-login-done-${version}.max.js`;
+  await fs.ensureDir(buildDir);
+  await fs.writeFile(`${buildDir}/${builtName}`, js);
```

Comment:
tip: i recently learned about `fs.outputFile`, which is the same as ensureDir + writeFile combined.

Reply:
Nice tip! I'm sure we can replace all the `fs.ensureDir` + `fs.writeFile` combos in this file and other files, but we'll still need to figure out what to do with the instances that use `fs.copy` in a loop, for example.

I'm planning on doing further cleanup once I tackle the file streaming in `compile.js`, in case it allows for the simplification of some of this code. Okay to punt on this until then?

Reply:
Did this and removed every useless call to `ensureDir`.